spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -91,79 +98,243 @@ public void setTaskExecutor(TaskExecutor taskExecutor) {
 	 * @param consumersPerQueue the consumers per queue.
 	 */
 	public void setConsumersPerQueue(int consumersPerQueue) {
+		if (isRunning()) {
+			adjustConsumers(consumersPerQueue);
+		}
 		this.consumersPerQueue = consumersPerQueue;
 	}
 
 	/**
-	 * Set to false to requeue failed deliveries.
-	 * @param requeueRejected the requeue rejected.
+	 * Set to true for an exclusive consumer - if true, the
+	 * {@link #setConsumersPerQueue(int) consumers per queue} must be 1.
+	 * @param exclusive true for an exclusive consumer.
 	 */
-	public void setDefaultRequeueRejected(boolean requeueRejected) {
-		this.defaultRequeueRejected = requeueRejected;
+	@Override
+	public final void setExclusive(boolean exclusive) {
+		Assert.isTrue(!exclusive || (this.consumersPerQueue == 1),
+				"When the consumer is exclusive, the consumers per queue must be 1");
+		super.setExclusive(exclusive);
 	}
 
 	@Override
+	public void setQueueNames(String... queueName) {
+		Assert.state(!isRunning(), "Cannot set queue names while running, use add/remove");
+		super.setQueueNames(queueName);
+	}
+
+	@Override
+	public void setQueues(Queue... queues) {
+		Assert.state(!isRunning(), "Cannot set queue names while running, use add/remove");
+		super.setQueues(queues);
+	}
+
+	@Override
+	public void addQueueNames(String... queueNames) {
+		addQueues(Arrays.asList(queueNames));
+		super.addQueueNames(queueNames);
+	}
+
+	@Override
+	public void addQueues(Queue... queues) {
+		addQueues(Arrays.asList(queues)
```

Comment:
Thank you, @kartoffelsup , for pointing that out!

Looking at this further I even think that we don't need those `.collect(Collectors.toList())`.
We can just get `Stream` here and in the previous `addQueueNames()` method and just `.map()` each item.
Although need to understand @garyrussell 's logic in the `addQueues(Collection<String> queues)` regarding double iteration over the same collection....